### PR TITLE
Fixed nesting of values and errors when the original values are nested

### PR DESCRIPTION
**Pull request by robertmaldon**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -32,37 +42,76 @@ function validate(validators, filteredValue) {
...
-      if (schemaItem.children) {
-        createValidations(schemaItem.children, value);
-      }
       promises.push(validateThing(value));
     }
   });
 }
 
+/**
+ * Adds a value with a nested key (dot notation) to an object, creating
```
Comment (jameslnewell) [inline lib/index.js]: Does this need tests? Should we use something like [object-path](https://www.npmjs.com/package/object-path) which is already has a comprehensive test suite?

File: README.md [hunk 2]
```diff
@@ -179,9 +181,39 @@ schema = {
...
+
+validator.validate(schema, values2).then(result => {
+    console.log(result.valid);  //false
+    console.log(result.values); //{hospitalProduct: {code: {extrasCode: '12' }}}
+    console.log(result.errors); //{hospitalProduct: {code: {abc: 'abc must be less than 2 chars'}}}
+});
 ```
 ##Note about ```filters``` and ```children```
-Filters and children are currently incombatible with each other.
+Filters and children are currently incompatible with each other.
```
Comment (jameslnewell) [inline README.md:216]: Do we know why?

Comment (robertmaldon) [inline README.md:216]: I'm not sure why. I'll ask Tom when he comes back from vacation.

**Comment by robertmaldon** [comment]
As suggested I replaced custom object hierarchy code with well used object-path module.

File: README.md (hunk 2 above)
Comment (ramesius) [inline README.md:216]: I originally wrote that part of the read me.
It just means that we currently do not evaluate whole object filters as part of the following execution path:
https://github.com/robertmaldon/schema-validator/blob/76a9e1728130a01bb34c0805bc0fbbdd1ea7ef51/lib/index.js#L81

This probably just needs a re-wording, something identifying the conscious decision to not evaluate filters for child objects and maybe a todo for having children work with filters

**Pull request merged**